src/Datasource/EntityTrait.php

```diff
@@ -785,6 +785,38 @@ public function getDirty()
     {
         return array_keys($this->_dirty);
     }
+    
+    /**
+     * return value if property exists otherwise throw Exception
+     *
+     * @param string $property
+     * @return mixed
+     */
+    public function getOrFail($property)
+    {
+        $value = $this->get($property);
+        if (!$this->propertyExists($property) && $value === null) {
+            throw new \BadMethodCallException('Property "' . $property . '" not found.');
+        }
+
+        return $value;
+    }
+
+    /**
+     * Check if property exists in entity even if it is null
+     *
+     * @param string $property
+     * @return bool
+     */
+    public function propertyExists($property)
+    {
+        $hasProperty = ($this->has($property) OR $this->isDirty($property));
```

Comment:
I'm not sure how I feel about mixing `&&` and `OR` instead of `||`.

Reply:
The code sniffer is: https://travis-ci.org/cakephp/cakephp/jobs/272836802